pymongo/common.py

```diff
@@ -524,56 +525,77 @@ def validate_tzinfo(dummy, value):
     return value
 
 
-# journal is an alias for j,
-# wtimeoutms is an alias for wtimeout,
-URI_VALIDATORS = {
-    'replicaset': validate_string_or_none,
-    'w': validate_non_negative_int_or_basestring,
-    'wtimeout': validate_non_negative_integer,
-    'wtimeoutms': validate_non_negative_integer,
-    'fsync': validate_boolean_or_string,
-    'j': validate_boolean_or_string,
+# Dictionary where keys are the names of public URI options, and values
+# are lists of aliases for that option. Aliases of option names are assumed
+# to have been deprecated.
+URI_OPTIONS_ALIAS_MAP = {
+    'journal': ['j', ],
+    'wtimeoutms': ['wtimeout', ],
+    'tls': ['ssl', ],
+    'tlsallowinvalidcertificates': ['ssl_cert_reqs', ],
+    'tlsallowinvalidhostnames': ['ssl_match_hostname', ],
+    'tlscrlfile': ['ssl_crlfile', ],
+    'tlscafile': ['ssl_ca_certs', ],
+    'tlsclientcertfile': ['ssl_certfile', ],
+    'tlsclientkeypassword': ['ssl_pem_passphrase', ],
+    'tlsclientkeyfile': ['ssl_keyfile', ],
+}
+
+# Dictionary where keys are the names of URI options, and values
+# are functions that validate user-input values for that option. If an option
+# alias uses a different validator than its public counterpart, it should be
+# included here as a key, value pair.
```

Comment:
> If an option alias uses a different validator than its public counterpart, it should be included here as a key, value pair.

Is there an example of such an option?

Reply:
`tlsAllowInvalidCertificates` is the renamed analogue of `ssl_cert_reqs`. These options have different validators as `ssl_cert_reqs` must be either `ssl.CERT_NONE` or `ssl.CERT_REQUIRED` while `tlsAllowInvalidCertificates` needs to be either True or False. In such a situation, the validator method also maps the boolean `tlsAllowInvalidCertificates` value to the appropriate `ss.CERT_*` constant.
Would you like me to add this to the comment?